pandas/core/series.py

```diff
@@ -2077,8 +2077,14 @@ def __rmatmul__(self, other):
     def searchsorted(self, value, side='left', sorter=None):
         if sorter is not None:
             sorter = ensure_platform_int(sorter)
-        return self._values.searchsorted(Series(value)._values,
-                                         side=side, sorter=sorter)
+        if not is_extension_type(self._values):
```

Comment:
hmm, this is orphaning pandas.core.base.searchsorted e.g. Index.searchsorted calls this. can you fix this up? (and add an asv for same). I think ok to make this if clause a helper function, maybe in algos.py and just call it in both places.

Reply:
Great idea  consolidating the two methods, I'll do that.

Now, there won't be a speed difference for Int64Index from this. But Uint64Index will benefit from a common implementation, as it today also experiences the casting problem:

```python
>>> i = pd.Index([1] * n + [2] * n + [3] * 3, dtype='uint64')
>>> %timeit i.searchsorted(2)
1.11 ms ± 4.98 µs per loop
```

So yeah, a common implentation is a great idea. I'll get to it.